flytepropeller/pkg/resolver/k8s_resolver.go

```diff
@@ -0,0 +1,182 @@
+	if len(parts) >= 2 {
+		service, namespace = parts[0], parts[1]
+	}
+
+	return
+}
+
+func parseResolverTarget(target resolver.Target) (targetInfo, error) {
+	var service, port, namespace string
+	if target.URL.Host == "" {
+		// kubernetes:///service.namespace:port
+		service, port, namespace = splitServicePortNamespace(target.Endpoint())
+	} else if target.URL.Port() == "" && target.Endpoint() != "" {
+		// kubernetes://namespace/service:port
+		service, port, _ = splitServicePortNamespace(target.Endpoint())
+		namespace = target.URL.Hostname()
+	} else {
+		// kubernetes://service.namespace:port
+		service, port, namespace = splitServicePortNamespace(target.URL.Host)
+	}
+
+	if service == "" {
+		return targetInfo{}, fmt.Errorf("target %s must specify a service", &target.URL)
+	}
+
+	return targetInfo{
+		serviceName:      service,
+		serviceNamespace: namespace,
+		port:             port,
+	}, nil
+}
+
+// Build creates a new resolver for the given target, e.g. k8s:///flyteagent:flyte:8000.
+func (b *kubeBuilder) Build(target resolver.Target, cc resolver.ClientConn, opts resolver.BuildOptions) (resolver.Resolver, error) {
+	ti, err := parseResolverTarget(target)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	r := &kResolver{
+		target:    ti,
+		ctx:       ctx,
+		cancel:    cancel,
+		cc:        cc,
+		k8sClient: b.k8sClient,
+	}
+	go wait.Until(r.run, 0, ctx.Done())
+	return r, nil
+}
+
+// Scheme returns the scheme supported by this resolver.
+// Scheme is defined at https://github.com/grpc/grpc/blob/master/doc/naming.md.
+func (b *kubeBuilder) Scheme() string {
+	return b.schema
+}
+
+type kResolver struct {
+	target    targetInfo
+	ctx       context.Context
+	cancel    context.CancelFunc
+	cc        resolver.ClientConn
+	k8sClient *kubernetes.Clientset
+	// wg is used to enforce Close() to return after the watcher() goroutine has finished.
+	wg sync.WaitGroup
+}
+
+// ResolveNow is a no-op at this point.
+func (k *kResolver) ResolveNow(resolver.ResolveNowOptions) {
+	logger.Infof(k.ctx, "k8s resolver: resolveNow")
+}
+
+// Close closes the resolver.
+func (k *kResolver) Close() {
+	k.cancel()
+	k.wg.Wait()
+	logger.Infof(k.ctx, "k8s resolver: closed")
+}
+
+func (k *kResolver) resolve(e *v1.Endpoints) {
+	var newAddrs []resolver.Address
+	for _, subset := range e.Subsets {
+		port := k.target.port
+
+		for _, address := range subset.Addresses {
+			newAddrs = append(newAddrs, resolver.Address{
+				Addr:       net.JoinHostPort(address.IP, port),
+				ServerName: fmt.Sprintf("%s.%s", k.target.serviceName, k.target.serviceNamespace),
+				Metadata:   nil,
+			})
+		}
+	}
+	logger.Infof(k.ctx, "k8s resolver: resolved %d addresses", len(newAddrs))
+	logger.Infof(k.ctx, "k8s resolver: resolved addresses: %v", newAddrs)
+	err := k.cc.UpdateState(resolver.State{Addresses: newAddrs, ServiceConfig: k.cc.ParseServiceConfig("{\"loadBalancingConfig\": [{\"round_robin\":{}}]}")})
```

Comment:
I think we don't want to send service config from resolver but only addresses, right? Resolver doesn't need to care about how addressed will be used. `cc.NewAddress(newAddrs)` might be good enough.

Reply:
I removed it, but forgot to push it. will update it